api/types/registry/registry.go

```diff
@@ -102,3 +105,15 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
+	// ID is the local image digest
+	ID digest.Digest
```

Comment:
> Does adding an omitempty tag set as @aaronlehmann suggested avoid that issue?

No, this breaks the fundamental relationship. Each image could have multiple IDs. You'll have to model is as a collection of ids, but then you are basically doing what the manifest format is doing.